src/mob.rs

```diff
@@ -0,0 +1,87 @@
+use bevy::app::{App, Plugin, Update};
+use bevy::math::{Vec2, Vec3};
+use bevy::prelude::{
+    default, Color, Commands, Component, Query, Res, ResMut, Resource, Sprite, SpriteBundle, Time,
+    Timer, Transform, With,
+};
+use bevy::time::TimerMode;
+use bevy::window::{PrimaryWindow, Window};
+use rand::Rng;
+
+static MOB_SPAWN_TIME: f32 = 5.;
+
+pub struct MobPlugin;
+impl Plugin for MobPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, move_mob);
+        app.add_systems(Update, spawn_mobs_over_time);
+        app.init_resource::<MobSpawnTimer>();
+        app.add_systems(Update, tick_mob_spawn_timer);
```

Comment:
```suggestion
        app.add_systems(Update, (move_mob, spawn_mobs_over_time))
            .add_systems(Update, tick_mob_spawn_timer)
            .init_resource::<MobSpawnTimer>();
```

I think I would prefer the builder-pattern like style for consistency